src/services/navigator/helpers.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) Garuda Labs, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import * as Types from 'hyperview/src/types-legacy';
+import * as UrlService from 'hyperview/src/services/url';
+import { ANCHOR_ID_SEPARATOR } from './types';
+
+/**
+ * Get an array of all child elements of a node
+ */
+export const getChildElements = (element: Types.Element): Types.Element[] => {
+  const elements: Types.Element[] = [];
+  if (element?.childNodes?.length) {
+    for (let i = 0; i < element?.childNodes?.length; i += 1) {
+      const child: Types.Element = element.childNodes[i];
+      if (child.nodeType === Types.NODE_TYPE.ELEMENT_NODE) {
+        elements.push(child);
+      }
+    }
+  }
+  return elements;
+};
```

Comment:
more succinct version of this code:
```suggestion
export const getChildElements = (element: Types.Element): Types.Element[] =>
  (element.childNodes || []).filter(
    (child: Types.Element) => child.nodeType === Types.NODE_TYPE.ELEMENT_NODE,
  );

```

Reply:
While this is prettier, childNodes is of type NodeList<T> which doesn't support iteration or filtering.

https://github.com/Instawork/hyperview/blob/d4077db6950abac8fad28ae30a19991b97b62a5a/src/types.js#L224-L227

Reply:
Fixed with Array.From()